<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DeviceDetailDialog</class>
 <widget class="QDialog" name="DeviceDetailDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>650</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Device Details</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QWidget" name="headerWidget" native="true">
     <layout class="QHBoxLayout" name="headerLayout">
      <item>
       <widget class="QLabel" name="iconLabel">
        <property name="minimumSize">
         <size>
          <width>48</width>
          <height>48</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>48</width>
          <height>48</height>
         </size>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="pixmap">
         <pixmap/>
        </property>
        <property name="scaledContents">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="headerInfoLayout">
        <item>
         <widget class="QLabel" name="ipAddressLabel">
          <property name="font">
           <font>
            <pointsize>14</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>192.168.1.1</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="hostnameLabel">
          <property name="text">
           <string>hostname.local</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="headerSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="statusLabel">
        <property name="font">
         <font>
          <pointsize>10</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>‚óè Online</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="Line" name="headerSeparator">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="overviewTab">
      <attribute name="title">
       <string>Overview</string>
      </attribute>
      <layout class="QVBoxLayout" name="overviewTabLayout">
       <item>
        <widget class="QGroupBox" name="basicInfoGroup">
         <property name="title">
          <string>Basic Information</string>
         </property>
         <layout class="QFormLayout" name="basicInfoFormLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="labelIpAddress">
            <property name="text">
             <string>IP Address:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="valueIpAddress">
            <property name="text">
             <string>-</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="labelHostname">
            <property name="text">
             <string>Hostname:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLabel" name="valueHostname">
            <property name="text">
             <string>-</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="labelMacAddress">
            <property name="text">
             <string>MAC Address:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLabel" name="valueMacAddress">
            <property name="text">
             <string>-</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="labelVendor">
            <property name="text">
             <string>Vendor:</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLabel" name="valueVendor">
            <property name="text">
             <string>-</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="labelDeviceType">
            <property name="text">
             <string>Device Type:</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QLabel" name="valueDeviceType">
            <property name="text">
             <string>-</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="networkInfoGroup">
         <property name="title">
          <string>Network Information</string>
         </property>
         <layout class="QFormLayout" name="networkInfoFormLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="labelFirstSeen">
            <property name="text">
             <string>First Seen:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="valueFirstSeen">
            <property name="text">
             <string>-</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="labelLastSeen">
            <property name="text">
             <string>Last Seen:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLabel" name="valueLastSeen">
            <property name="text">
             <string>-</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="labelOpenPorts">
            <property name="text">
             <string>Open Ports:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLabel" name="valueOpenPorts">
            <property name="text">
             <string>-</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="overviewSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="portsTab">
      <attribute name="title">
       <string>Ports</string>
      </attribute>
      <layout class="QVBoxLayout" name="portsTabLayout">
       <item>
        <widget class="QTableWidget" name="portsTableWidget">
         <property name="editTriggers">
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
         <property name="alternatingRowColors">
          <bool>true</bool>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::SingleSelection</enum>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectRows</enum>
         </property>
         <property name="sortingEnabled">
          <bool>true</bool>
         </property>
         <column>
          <property name="text">
           <string>Port</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Protocol</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>State</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Service</string>
          </property>
         </column>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="metricsTab">
      <attribute name="title">
       <string>Metrics</string>
      </attribute>
      <layout class="QVBoxLayout" name="metricsTabLayout">
       <item>
        <layout class="QHBoxLayout" name="metricsMainLayout">
         <item>
          <widget class="QGroupBox" name="networkMetricsGroup">
           <property name="title">
            <string>Network Metrics</string>
           </property>
           <layout class="QFormLayout" name="networkMetricsFormLayout">
            <item row="0" column="0">
             <widget class="QLabel" name="labelLatency">
              <property name="text">
               <string>Latency:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLabel" name="valueLatency">
              <property name="text">
               <string>-</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="labelPacketLoss">
              <property name="text">
               <string>Packet Loss:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QLabel" name="valuePacketLoss">
              <property name="text">
               <string>-</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="labelJitter">
              <property name="text">
               <string>Jitter:</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QLabel" name="valueJitter">
              <property name="text">
               <string>-</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="labelQualityScore">
              <property name="text">
               <string>Quality Score:</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QLabel" name="valueQualityScore">
              <property name="text">
               <string>-</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="qualityGaugeGroup">
           <property name="title">
            <string>Quality Gauge</string>
           </property>
           <property name="minimumSize">
            <size>
             <width>250</width>
             <height>250</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>300</width>
             <height>300</height>
            </size>
           </property>
           <layout class="QVBoxLayout" name="qualityGaugeLayout">
            <property name="spacing">
             <number>0</number>
            </property>
            <property name="leftMargin">
             <number>10</number>
            </property>
            <property name="topMargin">
             <number>10</number>
            </property>
            <property name="rightMargin">
             <number>10</number>
            </property>
            <property name="bottomMargin">
             <number>10</number>
            </property>
            <item>
             <widget class="QWidget" name="qualityGaugeContainer" native="true">
              <property name="minimumSize">
               <size>
                <width>200</width>
                <height>200</height>
               </size>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="metricsSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="historyTab">
      <attribute name="title">
       <string>History</string>
      </attribute>
      <layout class="QVBoxLayout" name="historyTabLayout">
       <item>
        <widget class="QGroupBox" name="historyFilterGroup">
         <property name="title">
          <string>Time Range</string>
         </property>
         <layout class="QHBoxLayout" name="historyFilterLayout">
          <item>
           <widget class="QLabel" name="labelTimeRange">
            <property name="text">
             <string>Show:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="timeRangeComboBox">
            <item>
             <property name="text">
              <string>Last Hour</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Last 6 Hours</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Last 24 Hours</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Last 7 Days</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Last 30 Days</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="refreshHistoryButton">
            <property name="text">
             <string>Refresh</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="historyFilterSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QTableWidget" name="historyTableWidget">
         <property name="editTriggers">
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
         <property name="alternatingRowColors">
          <bool>true</bool>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::SingleSelection</enum>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectRows</enum>
         </property>
         <column>
          <property name="text">
           <string>Timestamp</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Event</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Description</string>
          </property>
         </column>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="diagnosticsTab">
      <attribute name="title">
       <string>Diagnostics</string>
      </attribute>
      <layout class="QVBoxLayout" name="diagnosticsTabLayout">
       <item>
        <widget class="QGroupBox" name="tracerouteGroup">
         <property name="title">
          <string>Traceroute</string>
         </property>
         <layout class="QVBoxLayout" name="tracerouteGroupLayout">
          <item>
           <layout class="QHBoxLayout" name="tracerouteControlLayout">
            <item>
             <widget class="QPushButton" name="runTracerouteButton">
              <property name="text">
               <string>Run Traceroute</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="cancelTracerouteButton">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>Cancel</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="tracerouteSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QTextEdit" name="tracerouteOutput">
            <property name="readOnly">
             <bool>true</bool>
            </property>
            <property name="placeholderText">
             <string>Traceroute output will appear here...</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="diagnosticsButtonsLayout">
         <item>
          <widget class="QGroupBox" name="mtuGroup">
           <property name="title">
            <string>MTU Discovery</string>
           </property>
           <layout class="QVBoxLayout" name="mtuGroupLayout">
            <item>
             <widget class="QPushButton" name="discoverMtuButton">
              <property name="text">
               <string>Discover MTU</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="mtuResultLabel">
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
              <property name="text">
               <string>MTU: -</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="bandwidthGroup">
           <property name="title">
            <string>Bandwidth Test</string>
           </property>
           <layout class="QVBoxLayout" name="bandwidthGroupLayout">
            <item>
             <widget class="QPushButton" name="testBandwidthButton">
              <property name="text">
               <string>Test Bandwidth</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="bandwidthResultLabel">
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
              <property name="text">
               <string>Speed: -</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="dnsGroup">
           <property name="title">
            <string>DNS Diagnostics</string>
           </property>
           <layout class="QVBoxLayout" name="dnsGroupLayout">
            <item>
             <widget class="QPushButton" name="dnsLookupButton">
              <property name="text">
               <string>DNS Lookup</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="dnsResultLabel">
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
              <property name="text">
               <string>Records: -</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Close</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>DeviceDetailDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>DeviceDetailDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
